torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -87,6 +87,14 @@ static Tensor wrapped_scalar_tensor(Scalar scalar) {
   return tensor;
 }
 
+Tensor correct_dtype_gradients(ScalarType self_st, Tensor gradient_result) {
+  if (!at::isComplexType(self_st) && gradient_result.is_complex()) {
+    // R -> C
+    return at::real(gradient_result);
+  }
+  return gradient_result;
+}
```

Comment:
"correct_dtype_gradients" seems like a very generic name (e.g. it could be mistaken for a function that handles float -> double dtype conversion). Also, it looks like this function will be used quite a lot in autograd formulas.

Tossing some quick ideas out there:
- "handle_r_to_c", "handle_real_to_complex"
- "maybe_real_part"

Reply:
It seems to me that `at::real` should noop on real tensors, similar to how `at::conj` is noop on real tensors too.

Reply:
@zou3519 makes sense.`handle_real_to_complex` should be clearer, I think.

@ezyang yeah I agree. The reason we disabled `at::real` for non-complex tensors before was because it would be weird to have real return a view for non-complex tensors and `at::imag` return a new tensor populated with zeros (which it was before). We can certainly only enable `at::real` for non-complex tensors if we want though.

Reply:
Sure, but now that `at::conj` is a no-op for real tensors, I think we should probably be OK with making `at::real` do the same as well. I don't care... too much about `imag`, I don't think it shows up in situations like this.